services/stats/memstats/stats.go

```diff
@@ -105,7 +107,7 @@ func (m *Measurement) Gauge(value interface{}) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
-	m.values = append(m.values, value.(float64))
+	m.values = append(m.values, cast.ToFloat64(value))
```

Comment:
Just curious, why did you use the library for casting ?

Reply:
We are sending `int` till now in Guage stats.

Reply:
I think leo was asking, why add one more dependency. Why not in this case write the code yourself to convert to `float64` ?

Reply:
I think I misunderstood the comment. Since the value type is interface{} and it can be anything. So I thought of un-necessary handling the type casting by ourselves whenever the new type is being added we can use cast library.

Reply:
```Go
	var v float64

	switch value.(type) {
	case int:
		v = float64(value.(int))
	case float64:
		v = value.(float64)
	default:
		panic("unsupported type for Gauge")
	}
	m.values = append(m.values, v)
```
Should I go ahead with this one?
@lvrach @abhimanyubabbar

Reply:
> Should I go ahead with this one?
For me, the current approach is ok. I just wanted to understand the problem.